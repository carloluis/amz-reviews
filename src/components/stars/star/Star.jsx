import React from 'react';
import PropTypes from 'prop-types';
import { classes } from '../../../utils';
import styles from './Star.scss';

class Star extends React.PureComponent {
    render() {
        const { checked, number, onToggle } = this.props;

        return (
            <span className={classes(styles.container, checked && styles.checked)} onClick={() => onToggle(number)}>
                <i className={styles.square} />
                <span>{number} â˜…</span>
            </span>
        );
    }
}

Star.propTypes = {
    checked: PropTypes.bool.isRequired,
    number: PropTypes.number.isRequired,
    onToggle: PropTypes.func.isRequired
};

export default Star;
