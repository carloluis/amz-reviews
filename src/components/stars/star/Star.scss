@import '../../../styles/palette.scss';

.container {
    position: relative;
    display: inline-flex;
    justify-content: space-between;
    user-select: none;
    cursor: pointer;
    font-size: 16px;
    margin: 8px 4px;
    letter-spacing: 1px;

    &:first-child {
        margin-left: 0;
    }

    .square {
        display: inline-block;
        position: relative;
        width: 24px;
        height: 24px;
        border: 2px solid $blue;
        border-radius: 8px;
        margin-right: 4px;
    }

    &.checked {
        .square::after {
            content: '';
            display: inline-block;
            width: 16px;
            height: 16px;
            position: absolute;
            background: $blue;
            left: 2px;
            top: 2px;
            border-radius: 4px;
        }
    }
}
