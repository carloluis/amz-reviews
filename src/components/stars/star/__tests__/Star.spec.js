import React from 'react';
import renderer from 'react-test-renderer';
import Star from '../Star';

describe('<Star/>', () => {
    it('snapshot', () => {
        const result = renderer.create(<Star number={1} checked onToggle={jest.fn()} />).toJSON();
        expect(result).toMatchSnapshot();
    });
});
