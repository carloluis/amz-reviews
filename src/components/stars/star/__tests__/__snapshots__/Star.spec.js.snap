// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Star/> snapshot 1`] = `
<span
  className="container checked"
  onClick={[Function]}
>
  <i
    className="square"
  />
  <span>
    1
     â˜…
  </span>
</span>
`;
